treebeard/mp_tree.py

```diff
@@ -373,8 +373,11 @@ def process(self):
                       ' increasing the path.max_length property'
                       ' and UPDATE your database'))
         else:
-            # adding the new child as the last one
-            newobj.path = self.node.get_last_child()._inc_path()
+            # adding the new child as the last one after checking if there is a Last Child
+            if self.node.get_last_child():
```

Comment:
No need to call get_last_child twice

Reply:
Had been called Only once, is there another call of it that I am not aware of?

Reply:
my changes can become like:
newobj.path = [self.node.get_last_child()._inc_path() if self.node.get_last_child() else self.node.path + '0001']

Reply:
That's still two calls...

Reply:
I'm not aware how to make it one, please, feel free to modify